gh-actions/commit-size/action.yaml

```
@@ -0,0 +1,25 @@
+name: 'Commit Size'
+description: 'Make sure no commit in the pull request goes over a certain size'
+inputs:
+  size:
+    description: 'Total maximum size of commit (lines added+removed).'
+    required: false
+    default: '500'
+runs:
+  using: "composite"
+  steps:
+    - name: Make sure the size of each commit doesn't pass the desired size
+      shell: bash
+      run: |
+        set -e
+        git fetch origin refs/pull/${{ github.event.pull_request.number}}/head
+        commits_range=${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}
+        sum_commit() { git diff --numstat ${1} ${1}^ | cut -f1,2 -d$'\t' | tr $'\t' '+' | paste -sd+ - | bc; }
+        for commit in $(git log --pretty="format:%H" ${commits_range}); do
+            if [[ $(sum_commit ${commit}) -gt ${{ inputs.size }} ]]; then
+                 echo "Commit ${commit} is bigger than allowed size ${{ inputs.size }}."
+                 exit 1
+            fi
+
+            echo "Commit ${commit} is within allowed size ${{ inputs.size }}."
```

Also print size here?

```suggestion
            echo "Size of commit ${commit} (${size})  is within allowed size ${{ inputs.size }}."
```